crates/circuits/dec_bfv/src/circuit.rs

```diff
@@ -27,22 +27,32 @@ pub struct DecBfvCircuit {
     /// This determines whether to generate secret key shares (SecretKey) or
     /// smudging error shares (SmudgingNoise) when creating sample decryption data.
     pub sample_type: SampleType,
+    /// The security parameter (λ) this circuit is configured with
+    ///
+    /// This value is explicitly stored and can be used for fhe.rs function calls
+    /// that require the security parameter. Parameters are considered secure
+    /// if lambda >= 80, and insecure if lambda < 80.
+    pub security_parameter: usize,
 }
 
 impl DecBfvCircuit {
-    /// Create a new DecBfvCircuit instance with the specified sample type
+    /// Create a new DecBfvCircuit instance with the specified sample type and security parameter
     ///
     /// # Arguments
     ///
     /// * `sample_type` - The sample type (SecretKey or SmudgingNoise)
-    pub fn new(sample_type: SampleType) -> Self {
-        DecBfvCircuit { sample_type }
+    /// * `lambda` - The security parameter (λ)
+    pub fn new(sample_type: SampleType, lambda: usize) -> Self {
+        DecBfvCircuit {
+            sample_type,
+            security_parameter: lambda,
+        }
     }
 }
 
 impl Default for DecBfvCircuit {
     fn default() -> Self {
-        Self::new(SampleType::SecretKey)
+        Self::new(SampleType::SecretKey, 80) // Default to secure lambda
```

Comment:
do we need a constant like secure_lambda = 80 we can use everywhere so if that ever changes (for whatever reason) it's easier to update?